PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.webapp/src/com/microsoft/azuretools/webapp/ui/AppServiceCreateDialog.java

```diff
@@ -1348,28 +1371,44 @@ private void collectData() {
             index = comboAppServicePlanLocation.getSelectionIndex();
             model.setRegion(index < 0 ? null : binderAppServicePlanLocation.get(index).name());
 
-            index = comboAppServicePlanPricingTier.getSelectionIndex();
-            model.setPricing(index < 0 ? null : binderAppServicePlanPricingTier.get(index).toString());
+            model.setPricing(Optional.ofNullable(getSelectedPricingTier()).map(PricingTier::toString).orElse(APPSETTINGS_TOOLTIP));
         } else {
             index = comboAppServicePlan.getSelectionIndex();
             model.setAppServicePlanId(index < 0 ? null : binderAppServicePlan.get(index).id());
         }
 
         // Runtime
-        chooseWin = btnOSGroupWin.getSelection();
-        model.setOS(chooseWin ? OperatingSystem.WINDOWS : OperatingSystem.LINUX);
-        if (chooseWin) {
+        final OperatingSystem os = getSelectedOS();
+        model.setOS(getSelectedOS());
+        if (os == OperatingSystem.WINDOWS) {
```

Comment:
equals?

Reply:
Refers https://stackoverflow.com/questions/1750435/comparing-java-enum-members-or-equals, `==` could work as well as `equals` but also avoid NPE and add type compatibility check at compile time